src/main/java/frc/robot/commands/translationCommand.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class translationCommand extends CommandBase {
+  ProfiledPIDController pidControlX;
+  double currentX;
+  double desiredX;
+  double deltaX;
+
+
+  ProfiledPIDController pidControlY;
+  double currentY;
+  double desiredY;
+  double deltaY;
+
+  
+  /** Creates a new translation. */
+  public translationCommand( double deltaX, double deltaY) {
+    pidControlX = new ProfiledPIDController(0.1, 0.0, 0.0, 
+                                           new TrapezoidProfile.Constraints(3, 2));
+    pidControlY = new ProfiledPIDController(0.1, 0.0, 0.0, 
+                                           new TrapezoidProfile.Constraints(3,2));
+
+    this.deltaX = deltaX;
+    this.deltaY = deltaY;
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(SwerveSubsystem.getInstance());
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    //from odometry get currentX and currentY
+    currentX = SwerveSubsystem.getInstance().getPose().getX();
+    currentY = SwerveSubsystem.getInstance().getPose().getY();
+
+    desiredX = currentX + deltaX;
+    desiredY = currentY + deltaY;
+
+    //set the tolerance
+    pidControlX.setTolerance(0.1, 0.1);
+    pidControlY.setTolerance(0.1, 0.1);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    double x = pidControlX.calculate(currentX, desiredX);
+    double y = pidControlY.calculate(currentY, desiredY);
+
+    SwerveSubsystem.getInstance().drive(x, y, 0, true, true);
```

Comment:
Ideally, you could also add PID control of the heading. It's possible that it would slip a bit and rotate while rying to climb up the slippery surface.

It could probably use the same PID tunings with other commands like the one Khalil is working on (not tested yet, hopefully we can test it Saturday).

Reply:
The angle rotation will be determined during the test.